bevy_lint/docs/release.md

```diff
@@ -0,0 +1,51 @@
+# Release Checklist
+
+## Kick-off Pull Request
+
+1. Review the [changelog](../CHANGELOG.md) and ensure that all notable changes have been documented.
+2. Replace `[Unreleased]` heading with the version with the format `[X.Y.Z] - YYYY-MM-DD`.
+3. Update the `**All Changes**` link to compare from `main` to the new tag `lint-vX.Y.Z`. (E.g. `lint-v0.1.0...main` to `lint-v0.1.0...lint-v0.2.0`.)
+4. Remove the `-dev` suffix from the version in [`Cargo.toml`](../Cargo.toml), and ensure [`Cargo.lock`](../../Cargo.lock) also updates.
+5. Commit all of these changes and open a pull request.
+6. Merge the PR once a core Bevy maintainer approves it with no outstanding issues from other contributors.
+      - This starts the release process, enacting a freeze on all other changes until the release has finished.
+
+## Release on Github
+
+1. [Create a new Github release](https://github.com/TheBevyFlock/bevy_cli/releases/new).
+2. Set the tag to `lint-vX.Y.Z`.
+3. Set the title to `` `bevy_lint` - vX.Y.Z``
+4. Paste the following into the release description:
+
+````markdown
+One-sentence summary of changes. What awesome features can we spotlight? What critical bugs were fixed?
```

Comment:
It might help to more clearly identify parts that need to be filled out/replaced. Maybe like `<One-sentence summary of changes>` or `[One-sentence summary of changes]`.

If we want to make use of comments, we could also do something like `<!-- One-sentence summary of changes -->`. This has the benefit of not rendering in case it's accidentally missed, and doesn't clutter up the preview as you work on it.

Not a big deal for this tiny snippet, but might be worth setting a style now for the future (e.g. issue templates).